### PR TITLE
Arpit/temp bor sync

### PR DESCRIPTION
**Pull request by temaniarpit27**
# Description

Fix issues raised by Alchemy for bor syncing

# Changes

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Changes only for a subset of nodes

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Nodes audience

In case this PR includes changes that must be applied only to a subset of nodes, please specify how you handled it (e.g. by adding a flag with a default value...)

# Checklist

- [x] I have added at least 2 reviewer or the whole pos-v1 team
- [ ] I have added sufficient documentation in code
- [ ] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

# Cross repository changes

- [ ] This PR requires changes to heimdall
    - In case link the PR here:
- [ ] This PR requires changes to matic-cli
    - In case link the PR here:

## Testing

- [ ] I have added unit tests
- [ ] I have added tests to CI
- [ ] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### Manual tests

Please complete this section with the steps you performed if you ran manual tests for this functionality, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: core/rawdb/bor_receipt.go [hunk 1]
```diff
@@ -114,8 +114,8 @@ func ReadBorReceipt(db ethdb.Reader, hash common.Hash, number uint64) *types.Rec
 		return nil
 	}
 
-	body := ReadBody(db, hash, number)
-	if body == nil {
+	body := HasBody(db, hash, number)
```
Comment (JekaMas) [inline core/rawdb/bor_receipt.go:116]: I read more code, let's revert this change.

File: internal/ethapi/api.go [hunk 2]
```diff
@@ -1453,15 +1453,23 @@ func newRPCPendingTransaction(tx *types.Transaction, current *types.Header, conf
 func newRPCTransactionFromBlockIndex(b *types.Block, index uint64, config *params.ChainConfig, db ethdb.Database) *RPCTransaction {
 	txs := b.Transactions()
 
-	borReceipt := rawdb.ReadBorReceipt(db, b.Hash(), b.NumberU64())
-	if borReceipt != nil {
-		tx, _, _, _ := rawdb.ReadBorTransaction(db, borReceipt.TxHash)
+	if index >= uint64(len(txs)+1) {
```
Comment (JekaMas) [inline internal/ethapi/api.go:1456]: Won't it suppress working with Bor transactions?

Comment (manav2401) [reply internal/ethapi/api.go:1456]: Earlier `if index == uint64(len(txs))` check wasn't there and hence we added check on top before even making a call. I think with this check being added, we can remove this `if index >= uint64(len(txs)+1)` totally. But to answer your question, the check if against `len(txs)+1` so it keeps a buffer incase state-sync transactions are present.

Comment (cffls) [reply internal/ethapi/api.go:1456]: The check also prevents out of range index, which could be malicious, from invoking ReadBorReceipt or ReadBorTransaction.

**Review by pratikspatil024 — APPROVED** [review]
